internal/engine/postgresql/pg_catalog.go

```diff
@@ -16559,38 +17023,47 @@ func genPGCatalog() *catalog.Schema {
 				{
 					Type: &ast.TypeName{Name: "double precision"},
 				},
+				{
+					Type: &ast.TypeName{Name: "interval"},
+				},
 			},
-			ReturnType: &ast.TypeName{Name: "double precision"},
+			ReturnType: &ast.TypeName{Name: "interval"},
 		},
 		{
-			Name: "percentile_disc",
+			Name: "percentile_cont",
 			Args: []*catalog.Argument{
 				{
-					Type: &ast.TypeName{Name: "double precision"},
+					Type: &ast.TypeName{Name: "double precision[]"},
+				},
+				{
+					Type: &ast.TypeName{Name: "interval"},
 				},
 			},
-			ReturnType: &ast.TypeName{Name: "anyelement"},
+			ReturnType: &ast.TypeName{Name: "interval[]"},
 		},
 		{
 			Name: "percentile_disc",
 			Args: []*catalog.Argument{
 				{
 					Type: &ast.TypeName{Name: "double precision[]"},
 				},
+				{
+					Type: &ast.TypeName{Name: "anyelement"},
+				},
```

Comment:
I think it is because there are special functions which have "normal" arguments, and "aggregated group" arguments.

https://www.postgresql.org/docs/9.4/functions-aggregate.html#FUNCTIONS-ORDEREDSET-TABLE

Reply:
Ya, I ended up writing the SQL to distinguish between the 2 types of arguments, and it does indeed fix `percentile_*` functions, but now the basic aggregates are failing. (MAX, SUM, etc).

I think there needs to be more work in the compiler to treat / lookup these functions differently based on the context in the query.  So not sure where to go from here.
https://github.com/kyleconroy/sqlc/pull/1748/commits/c873256cc94227dd712083ac532595de77af9849